arangod/Aql/InAndOutRowExpressionContext.cpp

```diff
@@ -69,32 +69,27 @@ void InAndOutRowExpressionContext::invalidateInputRow() {
   _input = InputAqlItemRow{CreateInvalidInputRowHint{}};
 }
 
-bool InAndOutRowExpressionContext::isDataFromCollection(
-    Variable const* variable) const {
-  for (size_t i = 0; i < _vars.size(); ++i) {
-    auto const& v = _vars[i];
-    if (v->id == variable->id) {
-      if (variable->isDataFromCollection) {
-        return true;
-      }
-      TRI_ASSERT(i < _regs.size());
-      if (i == _vertexVarIdx || i == _edgeVarIdx || i == _pathVarIdx) {
-        return true;
-      }
-    }
-  }
-  return false;
-}
-
 AqlValue InAndOutRowExpressionContext::getVariableValue(
     Variable const* variable, bool doCopy, bool& mustDestroy) const {
   TRI_ASSERT(_input.isInitialized());
+
+  if (!_variables.empty()) {
+    TRI_ASSERT(_variables.empty());
+    auto it = _variables.find(variable);
+
+    if (it != _variables.end()) {
+      // copy the slice we found
+      mustDestroy = true;
+      return AqlValue((*it).second);
+    }
+  }
```

Comment:
5th occurence, see comments about moving it to protected `QueryExpressionContext` member above

Reply:
Done